app/src/main/java/com/android/sample/ui/workout/WorkoutScreen.kt

```diff
@@ -0,0 +1,405 @@
+                                          WorkoutType.RUNNING -> TODO()
+                                        }),
+                            contentDescription = "Goal Counter",
+                            modifier = Modifier.size(350.dp, 200.dp).testTag("ExerciseTypeIcon"))
+                      } else {
+                        CountDownTimer(
+                            timer,
+                            timeLimit,
+                            modifier =
+                                Modifier.size(220.dp).testTag("CountDownTimer").clickable {
+                                  countDownTimerIsPaused = !countDownTimerIsPaused
+                                },
+                            isPaused = countDownTimerIsPaused,
+                            countDownCurrentValue = countDownValue,
+                            isCountDownTime = isCountdownTime)
+                        Spacer(modifier = Modifier.height(5.dp))
+                      }
+                    }
+                  }
+
+              // Presentation button box
+              if (presentationButtonBoxIsDisplayed) {
+                Column(
+                    modifier = Modifier.size(height = 120.dp, width = 180.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.Bottom) {
+                      Button(
+                          onClick = { nextExercise() },
+                          colors = ButtonDefaults.buttonColors(containerColor = Color.Red),
+                          modifier = Modifier.testTag("SkipButton")) {
+                            Text("Skip", color = Color.Black)
+                          }
+                      Spacer(modifier = Modifier.height(25.dp))
+                      Button(
+                          onClick = {
+                            presentationButtonBoxIsDisplayed = false
+                            goalCounterBoxIsDisplayed = true
+                            finishButtonBoxIsDisplayed = true
+                            videoBoxIsDisplayed = false
+                          },
+                          modifier = Modifier.width(200.dp).height(50.dp).testTag("StartButton"),
+                          colors = ButtonDefaults.buttonColors(containerColor = Color(0xFFA9B0FF)),
+                          shape = RoundedCornerShape(size = 11.dp)) {
+                            Text("Start", color = Color.Black, fontSize = 20.sp)
+                          }
+                    }
+              } else if (finishButtonBoxIsDisplayed) {
+                // Finish button box to be displayed during a exercise to be executing
+                Column(
+                    modifier = Modifier.size(height = 250.dp, width = 180.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally,
+                    verticalArrangement = Arrangement.Top) {
+                      Button(
```

Comment:
Here you define once again the Skip Button. In 
`if (presentationButtonBoxIsDisplayed) {...}` you already defined something similar. Maybe it would  be a good idea to create here a separate composable to improve a bit the modularity.

Reply:
Solved